neqo-client/src/main.rs

```diff
@@ -728,7 +712,10 @@ impl<'a> Handler<'a> {
         while let Some(event) = client.next_event() {
             match event {
                 Http3ClientEvent::AuthenticationNeeded => {
-                    client.authenticated(AuthenticationStatus::Ok, Instant::now());
+                    // TODO
+                    if !client.conn().state().connected() {
+                        client.authenticated(AuthenticationStatus::Ok, Instant::now());
+                    }
                 }
```

Comment:
I need to further debug this. Seems like `AuthenticationNeeded` is emitted twice. First call to `client.authenticated` succeeds. Second call panics in QUIC Interop `http3` testcase. See second `Authenticated Ok` below.

```
client  | 0s 58ms INFO [Client 10e6071890efef92368199e561655e39a1] Authenticated Ok
client  | 0s 58ms DEBUG Write secret available for 3: SymKey [32]: a2d666f105371869843f35524206228b24c7b41e6251c9a8b42cb73257c6e009
client  | 0s 58ms DEBUG Read secret available for 3: SymKey [32]: ae05f06efb0dfccd16915d78387b3422652e3208e64ee4b6aa6b8d719a606fca
client  | 0s 58ms DEBUG [Agent 0x5654fe97cd50] SSL_AuthCertificateComplete: Ok(())
client  | 0s 58ms INFO [Agent 0x5654fe97cd50] state -> Complete(SecretAgentInfo { version: 772, cipher: 4865, group: 29, resumed: false, early_data: false, ech_accepted: false, alpn: Some("h3"), signature_scheme: 2052 })
client  | 0s 58ms INFO [Client 10e6071890efef92368199e561655e39a1] TLS connection complete
client  | 0s 58ms INFO Making Write 3 CryptoDxState, v=Version1 cipher=4865
client  | 0s 58ms DEBUG [Crypto] Application write key installed
client  | 0s 58ms INFO Making Read 3 CryptoDxState, v=Version1 cipher=4865
client  | 0s 58ms DEBUG [Crypto] application read keys installed
client  | 0s 59ms INFO [Client 10e6071890efef92368199e561655e39a1] State change from Handshaking -> Connected
client  | 0s 59ms INFO [Client 10e6071890efef92368199e561655e39a1] Connection established
client  | 0s 59ms INFO [Http3 connection] Fetch method=GET target: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("server4")), port: None, path: "/kofnudjjgb", query: None, fragment: None }
client  | 0s 59ms INFO [Client 10e6071890efef92368199e561655e39a1] Authenticated Ok
client  | thread 'main' panicked at neqo-crypto/src/agent.rs:621:9:
client  | assertion failed: self.state.authentication_needed()
client  | stack backtrace:
server  | 5s670ms INFO [Server 10e6071890efef92368199e561655e39a1] Drop packet number space in
server  | 5s670ms INFO [pri-path: [::]:443->[::ffff:193.167.0.100]:58001] discarding a packet without an RTT estimate; guessing RTT=48.642433ms
server  | 5s671ms INFO [Server 10e6071890efef92368199e561655e39a1] Rx ACK space=hs, ranges=[0..=0]
server  | 5s671ms INFO packet_acked this=0x55d0fb3f9660, pn=0, ps=1201, ignored=0, lost=0, rtt_est=RttEstimate { first_sample_time: Some(Instant { tv_sec: 66510, tv_nsec: 455582961 }), latest_rtt: 48.642433ms, smoothed_rtt: 48.642433ms, rttvar: 18.240912ms, min_rtt: 48.642433ms, ack_delay: Fixed(25ms) }
server  | 5s671ms INFO on_packets_acked this=0x55d0fb3f9660, limited=1, bytes_in_flight=195, cwnd=13370, state=SlowStart, new_acked=1201
server  | 5s671ms INFO Acked crypto frame space=hs offset=0 length=1161
server  | 5s671ms INFO TX blocked, profile=SendProfile { limit: 1337, pto: None, probe: (), paced: false } 
server  | 5s671ms INFO Setting timeout of 86.924772ms for socket 0
server  | 5s671ms INFO [Server 10e6071890efef92368199e561655e39a1] Rx ACK space=hs, ranges=[0..=1]
server  | 5s671ms INFO packet_acked this=0x55d0fb3f9660, pn=1, ps=195, ignored=0, lost=0, rtt_est=RttEstimate { first_sample_time: Some(Instant { tv_sec: 66510, tv_nsec: 455582961 }), latest_rtt: 35.47591ms, smoothed_rtt: 46.996617ms, rttvar: 16.972314ms, min_rtt: 35.47591ms, ack_delay: Fixed(25ms) }
server  | 5s671ms INFO on_packets_acked this=0x55d0fb3f9660, limited=1, bytes_in_flight=0, cwnd=13370, state=SlowStart, new_acked=195
server  | 5s671ms INFO Acked crypto frame space=hs offset=1161 length=154
server  | 5s671ms INFO TX blocked, profile=SendProfile { limit: 1337, pto: None, probe: (), paced: false } 
server  | 5s672ms INFO Setting timeout of 79.409963ms for socket 0
server  | 5s672ms INFO Setting timeout of 79.013306ms for socket 0
client  |    0: rust_begin_unwind
client  |              at ./rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:645:5
client  |    1: core::panicking::panic_fmt
client  |              at ./rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/panicking.rs:72:14
client  |    2: core::panicking::panic
client  |              at ./rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/panicking.rs:127:5
client  |    3: neqo_crypto::agent::SecretAgent::authenticated
client  |              at ./neqo/neqo-crypto/src/agent.rs:621:9
client  |    4: neqo_transport::connection::Connection::authenticated
client  |              at ./neqo/neqo-transport/src/connection/mod.rs:793:9
client  |    5: neqo_client::Handler::handle
client  |              at ./neqo/neqo-client/src/main.rs:715:21
client  |    6: neqo_client::ClientRunner::run::{{closure}}
client  |              at ./neqo/neqo-client/src/main.rs:870:17
client  |    7: neqo_client::main::{{closure}}
client  |              at ./neqo/neqo-client/src/main.rs:1136:18
client  |    8: tokio::runtime::park::CachedParkThread::block_on::{{closure}}
client  |              at ./usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.1/src/runtime/park.rs:282:63
client  |    9: tokio::runtime::coop::with_budget
client  |              at ./usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.1/src/runtime/coop.rs:107:5
client  |   10: tokio::runtime::coop::budget
client  |              at ./usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.1/src/runtime/coop.rs:73:5
client  |   11: tokio::runtime::park::CachedParkThread::block_on
client  |              at ./usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.1/src/runtime/park.rs:282:31
client  |   12: tokio::runtime::context::blocking::BlockingRegionGuard::block_on
client  |              at ./usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.1/src/runtime/context/blocking.rs:66:9
client  |   13: tokio::runtime::scheduler::multi_thread::MultiThread::block_on::{{closure}}
client  |              at ./usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.1/src/runtime/scheduler/multi_thread/mod.rs:87:13
client  |   14: tokio::runtime::context::runtime::enter_runtime
client  |              at ./usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.1/src/runtime/context/runtime.rs:65:16
client  |   15: tokio::runtime::scheduler::multi_thread::MultiThread::block_on
client  |              at ./usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.1/src/runtime/scheduler/multi_thread/mod.rs:86:9
client  |   16: tokio::runtime::runtime::Runtime::block_on
client  |              at ./usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.1/src/runtime/runtime.rs:350:50
client  |   17: neqo_client::main
client  |              at ./neqo/neqo-client/src/main.rs:1059:5
client  |   18: core::ops::function::FnOnce::call_once
client  |              at ./rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/ops/function.rs:250:5
client  | note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```